frontend/src/components/nav/workbench-nav.tsx

```diff
@@ -416,36 +429,78 @@ function WorkbenchNavOptions({
   workspaceId: string
   workflowId: string
 }) {
+
+  const router = useRouter()
+  const { deleteWorkflow } = useWorkflowManager()
+
+  const handleDeleteWorkflow = async () => {
+    console.log("Delete workflow")
+    await deleteWorkflow(workflowId)
+    router.push(`/workspaces/${workspaceId}`)
+    router.refresh()
```

Comment:
I don't think you have to refresh here, the `deleteWorkflow` should clear the react-query query key inside and show updated data

Reply:
I just copied the existing implementation:
https://github.com/TracecatHQ/tracecat/pull/328/files#diff-ffb1964b3632a8147cb3d858dfef3644c33d0e1493daca03283765543c6c53c6L36

Reply:
fairs, if it works without can we remove the refresh?

Reply:
I'll try

Reply:
Confirm works without